examples/seq2seq/finetune_trainer.py

```diff
@@ -322,7 +330,12 @@ def main():
     if training_args.do_predict:
         logger.info("*** Predict ***")
 
-        test_output = trainer.predict(test_dataset=test_dataset, metric_key_prefix="test")
+        test_output = trainer.predict(
+            test_dataset=test_dataset,
+            metric_key_prefix="test",
+            max_target_length=data_args.test_max_target_length,
```

Comment:
Also it's somewhat confusing that generate/config uses `max_length` attribute but trainer `max_target_length` - if we are creating new args, why not match the core naming?

(and there is no min_length override, but that's another story)

Reply:
for s2s, `max_target_length` makes sense IMO since here we have two seqs source seq, and target seq. Just `max_length` might create some confusion.

Reply:
Just to clarify I'm talking about the arg name to `predict` which then calls `generate`.

In no way I was proposing to change the cl args. 

Are we on the same page?

Reply:
Aah, I see, sorry, I misread it.

Reply:
I agree with @stas00 here, I think it's cleaner if the arg is called `max_length`. `max_length` in `generate()` always corresponds to the max_length that is predicted which would never correspond to "`max_source_length`". Since `evaluate` and `predict` are very closely tied to `generate()`, I think it's nice if this is a bit reflected in the function's signature by calling the arg `max_length`.

Reply:
Also do we really need `val_max_length` and `test_max_length`? -> Does it make much sense to have different values for those two? I can see why one would have `data_args.val_beams` smaller than `data_args.eval_beams`, but for `max_length` as well?

Reply:
It was added for convenience, but not super important, it makes sense to keep just one `max_target_length`

Reply:
I haven't touched anything on the script except using `test_max_length` here, which was never used before. I thought it was just a mistake. Since there is a lot of debate on this and I'm not sure I understand what you want me to do, I will revert that and leave it as it was before. Then we can do some renaming in a separate PR (or please be clearer on what you want me to do :-) )

Reply:
Oh I think I get it, so all `max_target_length` -> `max_length` and use one `eval_max_length`, like `eval_num_beams`. Will push that.

Reply:
exactly, sorry for the confusing comments ;)